ocaml/xapi/xapi_pool.ml

```diff
@@ -118,6 +112,72 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
             )
         )
   in
+  let one_ip_configured_on_joining_cluster_network () =
```

Comment:
Overall this feels too complicated. I don't know all the conditions that need to be met, though. It's just an intuition.

Reply:
I understand this only works for the cluster with "pool_auto_join=true".

Reply:
>I understand this only works for the cluster with "pool_auto_join=true".

Why is that? I thought `pool_auto_join` is for hosts already in the pool but not in the cluster to join the cluster automatically, not for hosts not yet in the pool?